tooling-docker/src/test/java/org/jboss/eap/qe/ts/common/docker/MalformedDockerConfigTest.java

```diff
@@ -0,0 +1,57 @@
+package org.jboss.eap.qe.ts.common.docker;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.startsWith;
+
+
+public class MalformedDockerConfigTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testFailFastWithMalformedDockerCommand() throws Exception {
+        Docker containerWithInvalidVersion = new Docker.Builder("wildfly", "jboss/wildfly:InvalidVersion")
+                .setContainerReadyTimeout(10, TimeUnit.SECONDS) // shorten timeout as this should fail fast
+                .setContainerReadyCondition(() -> false) // it's expected that server never starts and fails fast thus return false
+                .build();
+
+        thrown.expect(Exception.class);
+        thrown.expectMessage(startsWith("Starting of docker container using command: \"docker run --name"));
+        thrown.expectMessage(endsWith("failed. Check that provided command is correct."));
+
+        containerWithInvalidVersion.start();
+    }
+
+    @Test
+    public void testHangingReadyCondition() throws Exception {
+        Docker containerWithInvalidVersion = new Docker.Builder("wildfly", "jboss/wildfly:18.0.0.Final")
```

Comment:
I would change this from `containerWithInvalidVersion` to `hangingForReadyConditionContainer` or something similar.

Reply:
what about `testContainerReadyTimeoutWithHangingCondition`?

Reply:
Hi, it looks like the name for a _condition_, actually, rather than for something that represents a Docker instance. Using the same principle that you followed in the previous method (`containerWith*`) I'd have gone with something similar, like `containerWithHangingReadyCondition`. That said, I wouldn't want for us to trip over variable names too much, here.

Reply:
`testContainerWithHangingReadyCondition` it is ;-)

Reply:
Now I get where we clashed... I meant the `Docker` variable name, while you addressed the method name. Anyway, if you don't want to change it, that's fine to me too.

Reply:
Right :-), I've changed it.